airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -173,14 +199,44 @@ private static class DataSourceBuilder {
     private String jdbcUrl;
     private int maximumPoolSize = 10;
     private int minimumPoolSize = 0;
-    // the default 30000 millisecond is sometimes not enough for the acceptance test
-    private long connectionTimeoutMs = 60000;
+    private long connectionTimeoutMs;
     private String password;
     private int port = 5432;
     private String username;
+    private static final long MILLISECONDS = 1000;
+    private static final String CONNECT_TIMEOUT_KEY = "connectTimeout";
+    private static final String CONNECT_TIMEOUT_DEFAULT = "60";
 
     private DataSourceBuilder() {}
 
+    /**
+     * Retrieves connectionTimeout value from connection properties in seconds, default minimum timeout
+     * is 60 seconds since Hikari default of 30 seconds is not enough for acceptance tests. In the case
+     * the value is 0, pass the value along as Hikari and Postgres use default max value for 0 timeout
+     * value
+     *
+     * NOTE: HikariCP uses milliseconds for all time values:
+     * https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby whereas Postgres is
+     * measured in seconds: https://jdbc.postgresql.org/documentation/head/connect.html
+     *
+     * @param connectionProperties custom jdbc_url_parameters containing information on connection
+     *        properties
+     * @return DataSourceBuilder class used to create dynamic fields for DataSource
+     */
+    public DataSourceBuilder getConnectionTimeoutMs(final Map<String, String> connectionProperties) {
+      final long connectionTimeoutSeconds;
+      if (connectionProperties != null) {
```

Comment:
This will never happen because in `AbstractJdbcSource#createDataSource` the argument that will be passed in will be at minimum an empty Map since the argument for `connectionProperties` is actually created with the method `AbstractJdbcSource#getConnectionProperties`

As such, the test case of `testCreatingDataSourceWithConnectonTimeoutNotSet` covers this scenario, where in that case of no `connectionTimeout` set it will default to the value of 60 seconds

Reply:
@Amruta-Ranade 

> Is this the same parameter as `connectTimeout` in the [Postgres doc](https://jdbc.postgresql.org/documentation/head/connect.html) or is this an Airbyte-specific new parameter?

The field that we're passing in is the same as the `Posgres` doc, the reason the current implementation doesn't work is because our connector to `PostgresSource` expects the string to be `connectionTimeout` whereas `Postgres` is finicky and uses `connectTimeout` so users should expect to follow what `Postgres` defines and this code change will make that mapping on behalf of the users

I mainly tagged you since this issue was not clear why our connector to `Postgres` wasn't respecting the parameter that `Postgres` defined but it seems like a minor change that it limited to `Postgres` and not any of the other connectors we're trying to get to GA: `MySQL` and `MSSQL` where they use `connectionTimeout`